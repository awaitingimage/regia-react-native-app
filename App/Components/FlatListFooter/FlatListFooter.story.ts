import { storiesOf } from "@storybook/react-native";
import * as React from "react";

import FlatListFooter from "./FlatListFooter";

storiesOf("FlatListFooter", module)
  .add("Default", () => (
    <FlatListFooter />
  ));