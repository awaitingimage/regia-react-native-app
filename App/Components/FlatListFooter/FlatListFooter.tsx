import * as React from "react";
import { GestureResponderEvent, Text, View } from "react-native";
import styles from "./Style";

interface Props {
  onPress: (event: GestureResponderEvent) => void;
}

const FlatListFooter: React.SFC<Props> = ({onPress}: Props) =>  (
  <View style={styles.container}>
    <Text style={styles.linkText} onPress={onPress}>Return to Top</Text>
  </View >
);

export default FlatListFooter;
