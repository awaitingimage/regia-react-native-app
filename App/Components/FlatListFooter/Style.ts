import { StyleSheet } from "react-native";
import { ApplicationStyles, Colors, Fonts, Metrics } from "../../Themes";

export default StyleSheet.create({
  container: {
    flex: 1,
    height: 125,
    alignItems: "center",
    justifyContent: "center",
    flexDirection: "column",
  },
  linkText: {
    ...Fonts.style.description,
    color: Colors.link,
    lineHeight: Fonts.lineHeights.h5,
    textDecorationLine: "underline",
  },
});
