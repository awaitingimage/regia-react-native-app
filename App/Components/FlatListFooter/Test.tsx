/// <reference types="@types/jest" />
import * as React from "react";
import * as renderer from "react-test-renderer";
import FlatListFooter from "./FlatListFooter";

test("FlatListFooter component renders correctly", () => {
  const tree = renderer.create(<FlatListFooter onPress={() => console.log("test click")} />).toJSON();
  expect(tree).toMatchSnapshot();
});
