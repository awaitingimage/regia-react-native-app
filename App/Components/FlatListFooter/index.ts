import FlatListFooter from "./FlatListFooter";

export default FlatListFooter;
