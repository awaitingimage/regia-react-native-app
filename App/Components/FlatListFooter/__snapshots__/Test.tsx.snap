// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatListFooter component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "height": 125,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "color": "#1a0dab",
        "fontFamily": "Raleway",
        "fontSize": 14,
        "lineHeight": 24,
        "textDecorationLine": "underline",
      }
    }
  >
    Return to Top
  </Text>
</View>
`;
